### PR TITLE
Remove redundant shake.yaml

### PR DESCRIPTION
**Pull request by leifmetcalf**
As of #1136 we no longer need the nix config in `shake.yaml`.

### CODE REVIEW

**Comment by power-fungus** [comment]
**This does not work** (as described in [my comment](https://github.com/haskell/haskell-ide-engine/pull/1136#issuecomment-473623859))

On a fresh clone, this fails with the following message (just tested)
```
Stack looks for packages in the directories configured in the 'packages' and 'extra-deps' fields defined in your stack.yaml
The current entry points to $PWD/submodules/HaRe/ but no .cabal or package.yaml file could be found there.
```
The reason is that the script finds the `stack.yaml` file and tries to execute this script in the environment specified there.

Beeing able to run the script even with uninitialized submodules is a very critical feature. Before, there were a lot of issues regarding build errors where the submodules where not initialized correctly

**Comment by fendor** [comment]
I agree, I dont think we need the `shake.yaml` anymore.

File: install.hs [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #!/usr/bin/env stack
 {- stack
-  --stack-yaml shake.yaml
-  --install-ghc
-  runghc
+  script
+  --resolver nightly-2018-12-17
```
Comment (fendor) [inline install.hs]: resolver should not be equal to a resolver in one of the `stack-$VER.yaml`'s. See #1116 and the respective fix #1137.
tldr, I request changing the resolver to `nightly-2018-12-15`

Comment (leifmetcalf) [reply install.hs]: Done!
